### ISSUE TITLE
The 'first' modifier is not working correctly in nested (one-to-many) nodes using PostgreSQL.

### ISSUE THREAD

**Issue by ficolo** [prisma/prisma#3097]
**Describe the bug**
The 'first' modifier is not working correctly in nested (one-to-many) nodes using PostgreSQL.

**To Reproduce**
1. Init a prisma instance selecting the Docker local installation option with PostgreSQL as database
2. Deploy this datamodel [User, University, Career datamodel](https://gist.github.com/ficolo/05c00e3ca7d0382003b4e3a29f42dde6)
```js
type User {
  id: ID! @unique
  name: String
  interestedIn: [University!]!
}

type University {
  id: ID! @unique
  name: String
  careers: [Career!]!
}

type Career {
  id: ID! @unique
  name: String
}
```
3. Load this data [example-data](https://gist.github.com/ficolo/0007ac2aa8d79e44510f0ce24345464c)
4. Try to get all the users, with all the universities they are interested in, but only including the first 10 careers for those universities
```js
query{
  users{
    name
    interestedIn{
      name
      careers(first: 10){
        name
      }
    }
  }
}
```

**Expected behavior**
It should return the only user (username=Federico) with the three loaded universities with the first 10 careers for each university, but is only returning the first 5 careers for Harvard, the first 5 for Cambridge, and the all 3 careers for Oxford.

**Screenshots**
![image](https://user-images.githubusercontent.com/2343039/45545895-816b9180-b813-11e8-89f1-8a9b64d30311.png)
**Versions (please complete the following information):**
 - OS:  `OS X High Sierra`
 - `prisma` CLI:  1.16.2
 - Prisma Server: 1.6.2
- DB: PostgreSQL 10.5

**Additional context**
I tried to reproduce this bug using MySQL and it works as expected.

**Comment by LawJolla** [comment]
Probably the same as https://github.com/prisma/prisma/issues/3040

**Comment by do4gr** [comment]
We just released 1.16.4 which should fix this issue.

**Comment by ficolo** [comment]
Thanks!

I tested and it works as expected.
